pkg/crdutil/crdutil.go

```diff
@@ -84,38 +92,47 @@ func EnsureCRDsCmd() {
 		log.Fatalf("Failed to create API extensions client: %v", err)
 	}
 
-	if err := walkCrdsDir(ctx, client.ApiextensionsV1().CustomResourceDefinitions()); err != nil {
+	if err := walkCRDs(ctx, client.ApiextensionsV1().CustomResourceDefinitions()); err != nil {
 		log.Fatalf("Failed to apply CRDs: %v", err)
 	}
 }
 
-// walkCrdsDir walks the CRDs directory and applies each YAML file.
-func walkCrdsDir(ctx context.Context, crdClient v1.CustomResourceDefinitionInterface) error {
-	for _, crdDir := range crdsDir {
+// walkCRDs walks the CRDs directory and applies each YAML file.
+func walkCRDs(ctx context.Context, crdClient v1.CustomResourceDefinitionInterface) error {
+	for _, crdDir := range files {
+		// We can skip the errors as it has been checked in initFlags.
+		parentDir, _ := os.Stat(crdDir)
 		// Walk the directory recursively and apply each YAML file.
 		err := filepath.Walk(crdDir, func(path string, info os.FileInfo, err error) error {
 			if err != nil {
 				return err
 			}
-			if info.IsDir() || filepath.Ext(path) != ".yaml" {
+			if info.IsDir() {
```

Comment:
Question: if `path` is a directory should we not decend into it if recursive is set?

Reply:
But we walk through the files in the directory already and we are only interested in files

Reply:
See this test https://github.com/NVIDIA/k8s-operator-libs/pull/58#issuecomment-2502284586

Reply:
I think that's fine. Thanks for clarifying.